### PR TITLE
[X86] Fix an assembler bug of CMPCCXADD.

### PR DESCRIPTION
**Pull request by FreddyLeaf**

### CODE REVIEW

File: llvm/lib/Target/X86/MCTargetDesc/X86MCCodeEmitter.cpp [hunk 1]
```diff
@@ -948,10 +948,10 @@ X86MCCodeEmitter::emitVEXOpcodePrefix(int MemOperand, const MCInst &MI,
     llvm_unreachable("Unexpected form in emitVEXOpcodePrefix!");
   case X86II::MRMDestMem4VOp3CC: {
     //  MemAddr, src1(ModR/M), src2(VEX_4V)
+    Prefix.setR(MI, CurOp++);
```
Comment (phoebewang) [inline llvm/lib/Target/X86/MCTargetDesc/X86MCCodeEmitter.cpp:951]: Don't understand the change. Is the `CurOp` the index of operands?
According to above comment and assemble test, the order in `MemAddr, src1(ModR/M), src2(VEX_4V)` was correct. Does the change make it `src1(ModR/M), MemAddr, src2(VEX_4V)`?

Comment (KanRobert) [reply llvm/lib/Target/X86/MCTargetDesc/X86MCCodeEmitter.cpp:951]: Yes, I think so. The comment at line 950 needs to be corrected too.

File: llvm/test/MC/X86/cmpccxadd-intel-64.s [hunk 2]
```diff
@@ -768,3 +768,6 @@
 // CHECK: encoding: [0xc4,0x62,0xa9,0xe4,0x8a,0x00,0xfc,0xff,0xff]
```
Comment (KanRobert) [inline llvm/test/MC/X86/cmpccxadd-intel-64.s:768]: Why was it encoded correctly before this PR? I think r9 needs R bit too.

Comment (FreddyLeaf) [reply llvm/test/MC/X86/cmpccxadd-intel-64.s:768]: // CHECK: cmpsxadd  %r10, %r9, 268435456(%rbp,%r14,8)
It happens `%r10 ` `wrongly` set the R bit. So I added a test of `%ecx, %r8d, (%rip)`

File: llvm/lib/Target/X86/MCTargetDesc/X86MCCodeEmitter.cpp (hunk 1 above)
Comment (FreddyLeaf) [reply llvm/lib/Target/X86/MCTargetDesc/X86MCCodeEmitter.cpp:951]: You are correct. addressed in [1e66d6c](https://github.com/llvm/llvm-project/pull/66748/commits/1e66d6cc36130cd9d66e2233b3a78c3ca4222721)

**Review by phoebewang — APPROVED** [review]
LGTM.

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**